app/middleware/renderer.js

```diff
@@ -1,7 +1,51 @@
+const log = require('../lib/logger');
+const messages = require('../lib/messages');
+
 function findHelp(req, res) {
   res.render('find-help');
 }
 
+function location(req, res) {
+  if (res.locals.locationHeading === null) {
```

Comment:
Using the fact `locationHeading` hasn't been set is a side effect of the location not being correct, this might change. It would be clearer to set a specific flag and use that to check if an error has occurred.